runtime-decompiler/src/main/java/org/jrd/backend/data/ArchiveManager.java

```diff
@@ -0,0 +1,58 @@
+package org.jrd.backend.data;
+
+import org.jrd.backend.communication.FsAgent;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+public class ArchiveManager {
+    File c;
+    String tmpdir = System.getProperty("java.ioo.tmpdir");
+    ArrayList<String> currentJars = new ArrayList<>();
+
+    public ArchiveManager(File c) {
+        this.c = c;
+    }
+
+    public boolean isClassInFile(String clazz) throws IOException {
+        ZipInputStream zis = new ZipInputStream(new FileInputStream(c));
+        if (!currentJars.isEmpty()) {
+            currentJars = new ArrayList<>();
+        }
+        currentJars.add(c.getName());
+        var ret = findClazz(zis, clazz);
+        if (!ret) {
+            currentJars.remove(c.getName());
+        }
+        return ret;
+    }
+
+    private boolean findClazz(ZipInputStream zis, String clazz) throws IOException {
```

Comment:
Shouldnt it maybe return some accurate uri or other "path" ?

Reply:
ArchiveManager will be operating only on one archive and you'll ever need to find only one class. This method goes through the nested archives and saves "relative path" inside the current file into the currentJars ArrayList. This is later used in unpacking will be used in packing later. The idea is that ArchiveManager will now where the file you want to work with is and only tells you if it's there or not.

Reply:
So maybe it should rather keep other archvive managers instances?